generator/tests/Google/Service/Generator/ResourceTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Google\Service\Generator;
+
+class ResourceTest extends \PHPUnit\Framework\TestCase {
+  private $emptyNode = [];
+
+  function testGetClassName() {
+    $names = ['bidders',
+              'accounts',
+              'filterSets',
+              'filteredBids',
+              'creatives'];
+    $r = new Resource($names, $empty_node);
+
+    $deep_node = [
+      'resources' => ['accounts' => [
+        'resources' => ['filterSets' => [
+          'resources' => ['filteredBids' => [
+            'resources' => ['creatives' => []]]]]]]]];
+    $rparent = new Resource(['bidders'], $deep_node);
+    $rdeep = $rparent->getAllResources()[4];
+
+    $this->assertEquals('BiddersAccountsFilterSetsFilteredBidsCreatives', $r->getClassName());
+    $this->assertEquals('BiddersAccountsFilterSetsFilteredBidsCreatives', $rdeep->getClassName());
+  }
+
+  // function getLiteralName() {
```

Comment:
Are you planning to implement thse tests too?

Reply:
Done.